### PR TITLE
Fix testing on training data

### PR DESCRIPTION
**Pull request by grofte**
Both train_ds and test_ds was being created from the original dataframe df instead of the shuffled and split data. I fixed that and added an evaluation on the training data so something similar doesn't happen again.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
Thanks for catching this, the changes look good. Please add the generate files.

**Comment by grofte** [comment]
Yeah sure, no problem.
Can you post a link to an explanation of what the generate file is and
where I might find it?

**Comment by fchollet** [comment]
```
python autogen.py add_example nlp/tweet-classification-using-tfdf
```

**Comment by grofte** [comment]
You were right, there was also some code missing for generating the logs. But it was present in the notebook so I could just grab it there.

`black` did not want to run when installed from your requirements.txt. Something about a problem with `click`. So I did not run `black`.

**Comment by grofte** [comment]
Some new images were generated from the notebook but I did not add them since they were untracked. I'm not sure if this was wrong or not.

**Comment by fchollet** [comment]
> Continuous integration / black (pull_request) Failing after 22s — black

Please run `black`. You just need to install the right version (version number is found in `requirements.txt`)

**Comment by grofte** [comment]
I am very lazy so I created the environment from the requirements.txt file. I notice that you have updated the `black` version 4 days ago. But even if I change `black` back to 19.10b0 I still get the error from `click`. Version 8.0.0 does work though (pip installed 8.1.3). So maybe you want to add that to the requirements.txt idk.
Everything should hopefully pass now.

File: examples/nlp/md/tweet-classification-using-tfdf.md [hunk 1]
```diff
@@ -347,26 +347,4582 @@ model_2.fit(train_ds)
...
+Training model...
+Model trained in 0:00:41.461477
+Compiling model...
 
-<keras.callbacks.History at 0x7f453f9349d0>
+[INFO kernel.cc:1176] Loading model from path /tmp/tmpsp7fmsyk/model/ with prefix 20297ba36a694abd
+[INFO abstract_model.cc:1248] Engine "GradientBoostedTreesQuickScorerExtended" built
+[INFO kernel.cc:1022] Use fast generic engine
 
+WARNING:tensorflow:AutoGraph could not transform <function simple_ml_inference_op_with_handle at 0x7fe0aaefb5b0> and will run it as-is.
```
Comment (fchollet) [inline examples/nlp/md/tweet-classification-using-tfdf.md:360]: There are lots of WARNING and INFO logs in the file. Please remove them, they're noise.

File: examples/nlp/md/tweet-classification-using-tfdf.md [hunk 2]
```diff
@@ -347,26 +347,4582 @@ model_2.fit(train_ds)
...
...ation=Evaluation(num_examples=None, accuracy=0.825373113155365, loss=0.8096389174461365, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=162, evaluation=Evaluation(num_examples=None, accuracy=0.825373113155365, loss=0.8101357221603394, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=163, evaluation=Evaluation(num_examples=None, accuracy=0.8268656730651855, loss=0.8095107078552246, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=164, evaluation=Evaluation(num_examples=None, accuracy=0.8268656730651855, loss=0.8108780980110168, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=165, evaluation=Evaluation(num_examples=None, accuracy=0.8268656730651855, loss=0.8121252059936523, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=166, evaluation=Evaluation(num_examples=None, accuracy=0.8298507332801819, loss=0.8118909001350403, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=167, evaluation=Evaluation(num_examples=None, accuracy=0.8313432931900024, loss=0.8114330768585205, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None))]
```
Comment (fchollet) [inline examples/nlp/md/tweet-classification-using-tfdf.md:400]: Doesn't really look like something that adds value for the reader. Should you really print it?

File: examples/nlp/md/tweet-classification-using-tfdf.md [hunk 3]
```diff
@@ -347,26 +347,4582 @@ model_2.fit(train_ds)
...
...ion=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=142, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=143, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=144, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=145, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=146, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None)), TrainLog(num_trees=147, evaluation=Evaluation(num_examples=None, accuracy=0.5686567425727844, loss=1.3642939329147339, rmse=None, ndcg=None, aucs=None, auuc=None, qini=None))]
```
Comment (fchollet) [inline examples/nlp/md/tweet-classification-using-tfdf.md:414]: Likewise

File: examples/nlp/md/tweet-classification-using-tfdf.md [hunk 4]
```diff
@@ -347,26 +347,4582 @@ model_2.fit(train_ds)
...
+Type: "GRADIENT_BOOSTED_TREES"
+Task: CLASSIFICATION
+Label: "__LABEL"
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+Input Features (512):
+	model/keras_layer/StatefulPartitionedCall:0.0
```
Comment (fchollet) [inline examples/nlp/md/tweet-classification-using-tfdf.md:452]: This summary is pure noise, no added falue for the reader. We should not print it.

File: examples/nlp/md/tweet-classification-using-tfdf.md [hunk 5]
```diff
@@ -347,26 +347,4582 @@ model_2.fit(train_ds)
...
+    
+<div class="k-default-codeblock">
+```
+No weights
+```
+</div>
+    
+<div class="k-default-codeblock">
+```
+Variable Importance: MEAN_MIN_DEPTH:
```
Comment (fchollet) [inline examples/nlp/md/tweet-classification-using-tfdf.md:975]: Same

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

File: examples/nlp/md/tweet-classification-using-tfdf.md (hunk 4 above)
Comment (grofte) [reply examples/nlp/md/tweet-classification-using-tfdf.md:452]: Does this have anything to do with me? Or are you talking to someone else?

Comment (fchollet) [reply examples/nlp/md/tweet-classification-using-tfdf.md:452]: This is content that's being added to the `md` file as part of your PR, so it has to do with you. I suggest editing the Python file to remove the corresponding statements.

**Comment by grofte** [comment]
They are out but they'll be generated again as soon as someone runs the notebook themselves.

I didn't author this example so I am not very comfortable changing anything beyond the obvious mistake in the data processing function (in general I don't know too much about decision forest models - like, I don't get why they are even using the tf.data.Dataset if it is always a requirement that they load all data into memory).

**Review by fchollet — APPROVED** [review]
Thanks you for the edits. LGTM.
